### DISCUSSION TITLE
Proxmox lxc updater in crontab?

### DISCUSSION THREAD

**Discussion by HolgerM1** [tteck/Proxmox#1122] [Q&A] [answered]
Hi,
is it possible to run the updater as cronjob every month?

**Comment by yodaphone** [comment]
just download the script and set it up in crontab to run it as per your needs

**Reply by HolgerM1** [reply]
As the script is interactive, I will have to modify it for running unattended. Anybody having that set up?

**Reply by yodaphone** [reply]
just download the sh script & remove these lines & add to crontab

```
while true; do
  read -p "This Will Update All LXC Containers. Proceed(y/n)?" yn
  case $yn in
  [Yy]*) break ;;
  [Nn]*) exit ;;
  *) echo "Please answer yes or no." ;;
  esac
done
```

**Reply by HolgerM1** [reply]
Thanks! How would I start the script? Sorry, I'm really a noob.

**Comment by tteck** [comment, marked as answer]
This script will add/remove a crontab schedule that updates all LXCs every Sunday at midnight.

```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/cron-update-lxcs.sh)"
```
To edit crontab: `crontab -e`

To exclude LXCs from updating, edit crontab and add `CTID` as shown in the example (-s 103 111)

```
0 0 * * 0 PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin /bin/bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/update-lxcs-cron.sh)" -s 103 111 >>/var/log/update-lxcs-cron.log 2>/dev/null
```
To view Cron Update LXCs log: `cat /var/log/update-lxcs-cron.log`

**Comment by mgrimace** [comment]
Is there a way to be notified or to automate if a container needs a restart after its weekly cron update?
